CHANGELOG.md

```diff
@@ -13,6 +13,10 @@ git logs & PR history.
 
 # [Unreleased](https://github.com/puppetlabs/beaker/compare/3.35.0...master)
 
+### Added
+
+- Glossary for project jargon in `docs/concepts/glossary.md`
```

Comment:
this isn't necessary on its own, but since there are other changes needed for this PR, it would be cool if this had a link to the doc itself in it so you could click through to it.